src/core/algorithms/nd/nd.h

```diff
@@ -40,6 +50,14 @@ class ND {
     [[nodiscard]] std::string ToShortString() const;
 
     [[nodiscard]] std::string ToLongString() const;
+
+    [[nodiscard]] std::tuple<std::vector<std::string>, std::vector<std::string>, WeightType>
+    ToNameTuple() const {
+        return std::make_tuple(GetLhsNames(), GetRhsNames(), GetWeight());
+    }
```

Comment:
After moving the `GetVerticalNames()` implementation of this method should be placed in .cpp file.